README.md

```diff
@@ -1,23 +1,127 @@
+The Spark Conduit is a conduit contract designed to be used within the Maker Allocation System. It implements the IAllocatorConduit interface, so it will be able to work within the constraints on the Allocation System design. There are two contracts in this repo:
 
-Allocators can cancel withdrawal intents by calling `cancelFundRequest()`.
+1. `SparkConduit`: Facilitates the movement of funds between the Maker Allocation System and the SparkLend protocol.
+2. `DaiInterestRateStrategy`: Calculates the interest rate that is to be paid by all borrowers of DAI through the SparkLend protocol.
 
-Calling `requestFunds()` twice will override the previous fund request instead of add to it.
+## Roles/Permissions
 
-## SparkConduit
+1. `wards`: The `wards` mapping tracks the administrative permissions on this contract. Admin can upgrade the contract and set key storage values.
+2. `operator`: The `operator` performs actions on behalf of a given `ilk` (SubDAO identifier in the Maker Allocation System). The operator can deposit, withdraw, request funds, and cancel fund requests in the Conduit. Onboarding and offboarding of `operator` actors is done by Maker admin in the core system.
 
-`SparkConduit` is used for a local instance of Spark Lend. All dependency values are immediately accessible. The contract is upgradable to facilitate changes to logic if needed.
+### Admin Configuration
 
-### Configuration
+1. `roles`: The roles contract to perform operator authentication.
+2. `registry`: Returns the `buffer` contract for a given ilk (source of funds).
+3. `subsidySpread`: The delta between the Base Rate and the Subsidy Rate. [RAY]
+
+## Functionality
+
+### `deposit`
+
+The `deposit` function is used to move funds from a given `ilk`'s `buffer` into the Conduit. From the Conduit, the funds are used to `supply` in the SparkLend Pool. The result is that:
+1. Funds are moved from the `buffer` to SparkLend's aToken for that asset.
+2. aTokens are minted and sent to the Conduit.
+3. The Conduit state to track the `ilk`'s portion of the aTokens in the Conduit is increased.
+
+<p align="center">
+  <img src="https://github.com/marsfoundation/spark-conduits/assets/44272939/2a7cf453-3a7b-4d04-a0cd-d390cfeb8ec2" height="500" />
+</p>
+
+### `withdraw`
+
+The `withdraw` function is used to `withdraw` funds from the SparkLend Pool into the Conduit. From the Conduit, the funds are sent to the `ilk`'s `buffer`. The result is that:
+
+1. Funds are moved from SparkLend's aToken for that asset to the `buffer`.
+2. The Conduit's aTokens corresponding to the underlying asset withdrawn are burned.
+3. The Conduit state to track the `ilk`'s portion of the aTokens in the Conduit is reduced.
+
+<p align="center">
+  <img src="https://github.com/marsfoundation/spark-conduits/assets/44272939/fd64b9e2-28f3-45c6-8deb-7d43283e9443" height="500" />
```

Comment:
`1. deposit` is incorrectly labelled in the diagram.

Reply:
Fixed